WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -16,16 +16,72 @@ def competition_message_for_user(competition, user)
     messages.join(' ')
   end
 
-  def announced_class(days_announced)
-    level = [Competition::ANNOUNCED_DAYS_WARNING, Competition::ANNOUNCED_DAYS_DANGER].select { |d| days_announced > d }.count
-    ["alert-danger", "alert-orange", "alert-green"][level]
+  private def days_before_competition(date, competition)
+    date ? (competition.start_date - date.to_date).to_i : nil
   end
 
-  def report_and_results_class(days)
+  private def days_after_competition(date, competition)
+    date ? (date.to_date - competition.end_date).to_i : nil
+  end
+
+  def announced_content(competition)
+    days_announced = days_before_competition(competition.announced_at, competition)
+    days_announced ? "#{pluralize(days_announced, "day")} before" : ""
+  end
+
+  def announced_class(competition)
+    days_announced = days_before_competition(competition.announced_at, competition)
```

Comment:
This seems a repeat of line 28. Maybe you could move it to a separate method/variable?